src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -218,6 +218,10 @@ internal static void EnsureNullableAttributeExists(CSharpCompilation compilation
                         {
                             diagnostics.Add(ErrorCode.ERR_BadParameterModifiers, modifier.GetLocation(), SyntaxFacts.GetText(SyntaxKind.RefKeyword), SyntaxFacts.GetText(SyntaxKind.InKeyword));
                         }
+                        else if (parameter.ExclamationToken.Kind() != SyntaxKind.None)
+                        {
+                            diagnostics.Add(ErrorCode.ERR_NullCheckingOnByRefParameter, parameter.ExclamationToken.GetLocation(), parameter);
+                        }
```

Comment:
It doesn't feel like this belongs here. And the same code appears in 3 places. 

Consider moving to the caller which has the `RefKind`:
```
var refKind = GetModifiers(...);
if (refKind != RefKind.None)
{
    diagnostics.Add(...);
}
```